pymbolic/imperative/utils.py

```diff
@@ -135,12 +136,33 @@ def get_node_attrs(stmt):
 
 # {{{ graphviz / dot interactive show
 
-def show_dot(dot_code):
+class DotOutput(enum.Enum):
+    """
+    Records the output format of :func:`show_dot`.
+
+    .. atttribute:: XWINDOW
+
+        Shows the graph as an
+        `X window <https://en.wikipedia.org/wiki/X_Window_System>`_
+
+    .. atttribute:: BROWSER
+
+        Shows the graph as an SVG file in the local system's default
+        web-browser.
+    """
+    XWINDOW = "xwindow"
+    BROWSER = "browser"
+
+
+def show_dot(dot_code, output_to=DotOutput.XWINDOW):
```

Comment:
I generally like the idea, but I have worries about the concrete implementation.

- Introducing your own enum instead of passing through the output format as a string is artificially limiting.
- This breaks compatibility with any graphviz that doesn't have `x11` (be that windows or mac, or just built without it).

Reply:
> Introducing your own enum instead of passing through the output format as a string is artificially limiting.

Fair, now only accepts string arguments

> This breaks compatibility with any graphviz that doesn't have x11 (be that windows or mac, or just built without it).

Fair as well, now via [`dot -T?`](https://graphviz.org/docs/outputs/) we check if x11 is one of the backends available.